### PR TITLE
integration-cli: rely less on CLI output format

### PR DESCRIPTION
**Pull request by thaJeztah**
Running these tests with a different version of the CLI caused some failures because the tests relied on the CLI's output format.

Although these tests should be rewritten to use the API directly, in the meantime this makes them slightly more reliable.

### CODE REVIEW

**Comment by thaJeztah** [comment]
@andrewhsu @tiborvass @arkodg PTAL

File: integration-cli/docker_cli_events_unix_test.go [hunk 1]
```diff
@@ -396,35 +396,28 @@ func (s *DockerDaemonSuite) TestDaemonEvents(c *testing.T) {
 	defer os.Remove(configFilePath)
 
 	daemonConfig := `{"labels":["foo=bar"]}`
-	fmt.Fprintf(configFile, "%s", daemonConfig)
-	configFile.Close()
+	_, err = fmt.Fprintf(configFile, "%s", daemonConfig)
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_unix_test.go]: nit (no need to update for this): lol, might as well just do `configFile.Write([]byte(daemonConfig))`

Comment (tiborvass) [reply integration-cli/docker_cli_events_unix_test.go]: actually most of these should just be `ioutil.WriteFile` but whatever

Comment (thaJeztah) [reply integration-cli/docker_cli_events_unix_test.go]: yeah, was thinking about that when writing, guess I was lazy 😂

**Comment by thaJeztah** [comment]
ah, boo! something is failing

File: integration-cli/docker_cli_events_unix_test.go [hunk 2]
```diff
@@ -396,35 +396,28 @@ func (s *DockerDaemonSuite) TestDaemonEvents(c *testing.T) {
...
+	assert.NilError(c, configFile.Close())
+	assert.NilError(c, err)
 	s.d.Start(c, fmt.Sprintf("--config-file=%s", configFilePath))
 
-	// Get daemon ID
-	out, err := s.d.Cmd("info")
+	daemonID, err := s.d.Cmd("info", "--format", "{.ID}")
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_unix_test.go]: `{{.ID}}`

File: integration-cli/docker_cli_events_unix_test.go [hunk 3]
```diff
@@ -396,35 +396,28 @@ func (s *DockerDaemonSuite) TestDaemonEvents(c *testing.T) {
...
-	daemonName := ""
-	for _, line := range strings.Split(out, "\n") {
-		if strings.HasPrefix(line, "ID: ") {
-			daemonID = strings.TrimPrefix(line, "ID: ")
-		} else if strings.HasPrefix(line, "Name: ") {
-			daemonName = strings.TrimPrefix(line, "Name: ")
-		}
-	}
 	assert.Assert(c, daemonID != "")
+	daemonName, err := s.d.Cmd("info", "--format", "{.Name}")
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_unix_test.go]: `{{.Name}}`

File: integration-cli/docker_cli_events_unix_test.go [hunk 4]
```diff
@@ -464,28 +457,21 @@ func (s *DockerDaemonSuite) TestDaemonEventsWithFilters(c *testing.T) {
...
-	fmt.Fprintf(configFile, "%s", daemonConfig)
-	configFile.Close()
+	_, err = fmt.Fprintf(configFile, "%s", daemonConfig)
+	assert.NilError(c, configFile.Close())
+	assert.NilError(c, err)
 	s.d.Start(c, fmt.Sprintf("--config-file=%s", configFilePath))
 
-	// Get daemon ID
-	out, err := s.d.Cmd("info")
+	daemonID, err := s.d.Cmd("info", "--format", "{.ID}")
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_unix_test.go]: `{{.ID}}`

File: integration-cli/docker_cli_events_unix_test.go [hunk 5]
```diff
@@ -464,28 +457,21 @@ func (s *DockerDaemonSuite) TestDaemonEventsWithFilters(c *testing.T) {
...
-	for _, line := range strings.Split(out, "\n") {
-		if strings.HasPrefix(line, "ID: ") {
-			daemonID = strings.TrimPrefix(line, "ID: ")
-		} else if strings.HasPrefix(line, "Name: ") {
-			daemonName = strings.TrimPrefix(line, "Name: ")
-		}
-	}
 	assert.Assert(c, daemonID != "")
-	assert.NilError(c, s.d.Signal(unix.SIGHUP))
+	daemonName, err := s.d.Cmd("info", "--format", "{.Name}")
```
Comment (tiborvass) [inline integration-cli/docker_cli_events_unix_test.go]: `{{.Name}}`

File: integration-cli/docker_cli_events_unix_test.go (hunk 2 above)
Comment (tiborvass) [reply integration-cli/docker_cli_events_unix_test.go]: You could make this `{{.ID}}|{{.Name}}` and then split it on `|` to only do one call to info.

Comment (thaJeztah) [reply integration-cli/docker_cli_events_unix_test.go]: LOL I had that initially

File: integration-cli/docker_cli_events_unix_test.go (hunk 3 above)
Comment (thaJeztah) [reply integration-cli/docker_cli_events_unix_test.go]: ah, crap thx

File: integration-cli/docker_cli_events_unix_test.go (hunk 2 above)
Comment (tiborvass) [reply integration-cli/docker_cli_events_unix_test.go]: I still think that would be better, but let me know if you disagree

File: integration-cli/docker_cli_events_unix_test.go (hunk 1 above)
Comment (tiborvass) [reply integration-cli/docker_cli_events_unix_test.go]: No worries

**Comment by tiborvass** [comment]
I updated the PR with two commits

**Review by andrewhsu — APPROVED** [review]
SGTM w.r.t. having newer CLI versions test well against the daemon

**Comment by andrewhsu** [comment]
Hmm...i see [failure](https://ci.docker.com/public/blue/organizations/jenkins/moby/detail/PR-40053/5/tests):
```
--- FAIL: TestDockerDaemonSuite/TestDaemonEvents (3.58s)
daemon.go:26: Creating a new daemon at: "/go/src/github.com/docker/docker/bundles/test-integration/TestDockerDaemonSuite/TestDaemonEvents"
docker_cli_events_unix_test.go:443: assertion failed: expression is false: strings.Contains(out, s)
```

**Comment by thaJeztah** [comment]
Ha! It's the trailing newline that's causing the failure;

```
docker_cli_events_unix_test.go:444: assertion failed: string "2019-10-09T17:04:12.450810959Z daemon reload E63N:AJ4I:KCNZ:ZJ6N:NGUY:FKPA:TC2F:WSBY:D32Q:JVVM:ZBK4:CKMO (allow-nondistributable-artifacts=[], cluster-advertise=, cluster-store=, cluster-store-opts=null, debug=true, default-cgroupns-mode=host, default-ipc-mode=private, default-runtime=runc, default-shm-size=67108864, features=map[], insecure-registries=[], labels=[\"bar=foo\"], live-restore=false, max-concurrent-downloads=1, max-concurrent-uploads=5, max-download-attempts=5, name=78dfe3013f66, registry-mirrors=[], runtimes=runc:{runc []}, shutdown-timeout=10)\n" does not contain " name=78dfe3013f66\n"
```

**Comment by thaJeztah** [comment]
@tiborvass @kolyshkin this is green now

**Review by tiborvass — APPROVED** [review]

**Comment by thaJeztah** [comment]
@AkihiroSuda ptal 🤗

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**